lib/motions/move-to-definition-motion.coffee

```diff
@@ -0,0 +1,40 @@
+_ = require 'underscore-plus'
+{Input} = require '../view-models/view-model'
+{SearchCurrentWord} = require './search-motion'
+
+module.exports =
+class MoveToDefinition extends SearchCurrentWord
+  constructor: (@editor, @vimState) ->
+    super(@editor, @vimState)
+
+    # FIXME: This must depend on the current language
+    # NOTE: The only modification here is the addition of the $
+    defaultIsKeyword = "[@$a-zA-Z0-9_\-]+"
+    userIsKeyword = atom.config.get('vim-mode.iskeyword')
+    @keywordRegex = new RegExp(userIsKeyword or defaultIsKeyword)
+
+    searchString = @getCurrentWordMatch()
+    @input = new Input(searchString)
+
+  getSearchTerm: (term) ->
+    search = "(\\w )?#{term}\\s?=?"
+    search = search.replace /([$])/g, '\\$1'
+    new RegExp search, 'gmi'
+
+  scan: (cursor) ->
+    actualRange = null
+    @editor.scan @getSearchTerm(@input.characters), (iteration) ->
+      actualRange = iteration.range
+
+      # if there are two filled matches, that means
+      # this followed the first and second part, so
+      # we need to skip the match forward a bit
+      if _.compact(iteration.match).length is 2
+        actualRange.start.column += 2
+
+      # scan starts from the top, so we just want
+      # the first result
+      iteration.stop()
+
+    return [actualRange] if actualRange
+    []
```

Comment:
Yeah, I can't remember why I had that bit in. It was compensating for the regex defined above, but I didn't leave any notes as to why I needed that! I'm also looking into simplifying the code to re-use more of the parent class. The only thing I'm noticing is that the super scan reads through the entire document without stopping whereas mine stops the first item it finds. I could modify the super function to take an optional parameter of something like `returnFirstResult`. See anything odd about trying that?